modules/dnn/src/dnn.cpp

```diff
@@ -1269,6 +1273,8 @@ struct Net::Impl
             allocateLayer(lid, layersShapes);
         }
 
+        layersTimings.resize(lastLayerId + 1, 0);
```

Comment:
The last one thing, method `getLayerNames` returns all layers except data layer but `getPerfProfile` not. So, it seems better for me if we skip data layer in both methods.